pkg/generate/app/componentref.go

```diff
@@ -299,6 +302,9 @@ type ComponentInput struct {
 
 	ExpectToBuild bool
 	ScratchImage  bool
+	// For keeping track of if this image was retrieved from a Dockerfile,
+	// which changes our BuildConfig results
```

Comment:
@bparees I was thinking about whether it should explain the intent instead of the idea, I'll change the comment

@rhcarvalho When the input [image]~[source] gets converted into a ComponentReference, and then into an ImageRef, the information about where the image came from gets lost.  If we just specify the [source], the image gets filled in from the Dockerfile "FROM".  I added a flag so that some information gets preserved, and then I use that information to change the outputted BuildConfig.  I really don't like the flow either, it's kinda scotch-tape, but couldn't find a better way to do it.  I'm definitely open to suggestions!